File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/visitor/hosting/HostingCalculatorVisitor.java
// HostingCalculatorVisitor.java

package com.bigboxcode.designpattern.visitor.hosting;

public interface HostingCalculatorVisitor {

    double visit(ComputeService computeService);

    double visit(ContainerService containerService);

    double visit(DatabaseService databaseService);

    double visit(FileStorageService fileStorageService);

    double visit(ServerlessService serverlessService);
}

File: design-patterns/ts/visitor/hosting-cost/hosting-calculator-visitor.ts
// hosting-calculator-visitor.ts

import ComputeService from "./compute-service";
import ContainerService from "./container-service";
import DatabaseService from "./database-service";
import FileStorageService from "./file-storage-service";
import ServerlessService from "./serverless-service";
import Service from "./service";

interface HostingCalculatorVisitor {
    visit(service: Service): number;
}

export default HostingCalculatorVisitor;